cpp/src/random_projection/rproj.cuh

```diff
@@ -87,9 +86,9 @@ void sparse_random_matrix(const raft::handle_t& h,
 
     for (size_t i = 0; i < params.n_components; i++) {
```

Comment:
Given that the components here can get quite high (unlike TSNE, for example, where it might be reasonable to assume n_components = 2), I think it would be worth also investigating the ability to increase the parallelism here (as a potential future task, of course).